cns/restserver/internalapi.go

```diff
@@ -14,3 +27,123 @@ func (service *HTTPRestService) GetPartitionKey() (dncPartitionKey string) {
+func (service *HTTPRestService) SetNodeOrchestrator(r *cns.SetOrchestratorTypeRequest) {
+	body, _ := json.Marshal(r)
+	req, _ := http.NewRequest(http.MethodPost, "", bytes.NewBuffer(body))
+	req.Header.Set("Content-Type", "application/json; charset=UTF-8")
+	service.setOrchestratorType(httptest.NewRecorder(), req)
+}
+
+// SyncNodeStatus :- Retrieve the latest node state from DNC & returns the first occurence of returnCode and error with respect to contextFromCNI
+func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string, contextFromCNI json.RawMessage) (returnCode int, errStr string) {
+	logger.Printf("[Azure CNS] SyncNodeStatus")
+	var (
+		response         *http.Response
+		err              error
+		nodeInfoResponse cns.NodeInfoResponse
+		req              *http.Request
+		body             []byte
+		httpc            = common.GetHttpClient()
+	)
+
+	// try to retrieve NodeInfoResponse from mDNC
+	response, err = httpc.Get(fmt.Sprintf(common.SyncNodeNetworkContainersURLFmt, dncEP, infraVnet, nodeID, dncApiVersion))
+	if err == nil {
+		if response.StatusCode == http.StatusOK {
+			err = json.NewDecoder(response.Body).Decode(&nodeInfoResponse)
+		} else {
+			err = fmt.Errorf("%d", response.StatusCode)
+		}
+
+		response.Body.Close()
+	}
+
+	if err != nil {
+		returnCode = UnexpectedError
+		errStr = fmt.Sprintf("[Azure-CNS] Failed to sync node with error: %+v", err)
+		logger.Errorf(errStr)
+		return
+	}
+
+	var (
+		ncsToBeAdded   = make(map[string]cns.CreateNetworkContainerRequest)
+		ncsToBeDeleted = make(map[string]bool)
+	)
+
+	// determine new NCs and NCs to be deleted
+	service.RLock()
+	for ncid := range service.state.ContainerStatus {
+		ncsToBeDeleted[ncid] = true
+	}
+
+	for _, nc := range nodeInfoResponse.NetworkContainers {
+		ncid := cns.SwiftPrefix + nc.NetworkContainerid
+		delete(ncsToBeDeleted, ncid)
+		if savedNc, exists := service.state.ContainerStatus[ncid]; !exists || savedNc.CreateNetworkContainerRequest.Version < nc.Version {
+			ncsToBeAdded[ncid] = nc
+		}
+	}
+	service.RUnlock()
+
+	// check if the version is valid and save it to service state
+	for ncid, nc := range ncsToBeAdded {
+		var (
+			versionURL = fmt.Sprintf(nodeInfoResponse.GetNCVersionURLFmt,
+				nmagentclient.WireserverIP,
+				nc.PrimaryInterfaceIdentifier,
+				nc.NetworkContainerid,
+				nc.AuthorizationToken)
+			w = httptest.NewRecorder()
+		)
+
+		ncVersionURLs.Store(nc.NetworkContainerid, versionURL)
+		waitingForUpdate, tmpReturnCode, tmpErrStr := isNCWaitingForUpdate(nc.Version, nc.NetworkContainerid)
+		if tmpReturnCode != Success && bytes.Compare(nc.OrchestratorContext, contextFromCNI) == 0 {
+			returnCode = tmpReturnCode
+			errStr = tmpErrStr
+		}
+
+		if tmpReturnCode == UnexpectedError {
+			continue
+		}
+
+		body, _ = json.Marshal(nc)
+		req, _ = http.NewRequest(http.MethodPost, "", bytes.NewBuffer(body))
+		req.Header.Set("Content-Type", "application/json; charset=UTF-8")
```

Comment:
i see this used in multiple places..can we move this to constant
"Content-Type", "application/json; charset=UTF-8"

Reply:
fixed